### PR TITLE
MGTM-3751 retry to send controller logs from installer and recheck co…

### PR DESCRIPTION
**Pull request by tsorya**
…ntroller state

### CODE REVIEW

File: src/installer/installer.go [hunk 1]
```diff
@@ -465,23 +469,29 @@ func (i *installer) waitForController() error {
...
 			// if we will see that it is not enough, we will change the place.
 			err = common.UploadPodLogs(kc, i.inventoryClient, i.ClusterID, controllerPod.Name, assistedControllerNamespace, common.ControllerLogsSecondsAgo, i.log)
 			// if failed to upload logs, log why and continue
 			if err != nil {
-				i.log.WithError(err).Warnf("Failed to upload controller logs")
+				if retryCounterUploadControllerLogs > 0 {
+					retryCounterUploadControllerLogs--
+					controllerPod = nil
+				}
+				i.log.WithError(err).Warnf("Failed to upload controller logs, left %d retries", retryCounterUploadControllerLogs)
```
Comment (eranco74) [inline src/installer/installer.go:484]:
```suggestion
				i.log.WithError(err).Warnf("Failed to upload controller logs, %d retries left", retryCounterUploadControllerLogs)
```

File: src/installer/installer.go [hunk 2]
```diff
@@ -465,23 +469,29 @@ func (i *installer) waitForController() error {
 		controllerPod := common.GetPodInStatus(kc, assistedControllerPrefix, assistedControllerNamespace,
 			map[string]string{"job-name": assistedControllerPrefix}, v1.PodRunning, i.log)
 		if controllerPod != nil {
+
 			// uploading logs here cause we will finish installing bootstrap in couple of seconds
 			// just didn't wanted to start handling this logic in other place
 			// controller must be running for at least couple of minutes before this code so it will cover 90% of the cases
 			// if we will see that it is not enough, we will change the place.
 			err = common.UploadPodLogs(kc, i.inventoryClient, i.ClusterID, controllerPod.Name, assistedControllerNamespace, common.ControllerLogsSecondsAgo, i.log)
```
Comment (eranco74) [inline src/installer/installer.go:477]: I think this method needs some refactoring as well.
The log says that we fail to upload while we suspect that we failed to get the log from kube-apiserver.

Comment (tsorya) [reply src/installer/installer.go:477]: this method returns error, in the case we saw:
"Failed to get logs of pod assisted-installer-controller-p7pjv: Get \"https://192.168.50.122:10250/containerLogs/assisted-installer/assisted-installer-controller-p7pjv/assisted-installer-controller?sinceSeconds=3600\": open /etc/kubernetes/secrets/kube-apiserver-to-kubelet-client.crt: no such file or directory"

Comment (tsorya) [reply src/installer/installer.go:477]: than we just wrap it with one more message that is relevant for our specific case. We can though just print the error that returns from UploadPodLogs

**Pull request merged**